src/main.rs

```diff
@@ -42,10 +50,13 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         ..Default::default()
     };
 
-    let mut discovery = Discovery::new(portalnet_config.clone()).unwrap();
-    discovery.start().await.unwrap();
+    // Initialize base discovery protocol
+    let discovery = Arc::new(RwLock::new(
+        Discovery::new(portalnet_config.clone()).unwrap(),
+    ));
+    discovery.write().await.start().await.unwrap();
 
-    // Initialize state sub-network, if selected
+    // Initialize state sub-network jsonrpc handler, if selected
     let (state_tx, state_handler) = if trin_config.networks.iter().any(|val| val == "state") {
```

Comment:
I'd recommend using the new constants for `"state"` here and `"history"` just below